src/harness/typeWriter.ts

```diff
@@ -97,7 +97,7 @@ class TypeWriterWalker {
         if (!isSymbolWalk) {
             // Don't try to get the type of something that's already a type.
             // Exception for `T` in `type T = something` because that may evaluate to some interesting type.
-            if (ts.isPartOfTypeNode(node) || ts.isIdentifier(node) && !(ts.getMeaningFromDeclaration(node.parent) & ts.SemanticMeaning.Value) && !(ts.isTypeAlias(node.parent) && node.parent.name === node)) {
+            if (ts.isPartOfTypeNode(node) || ts.isIdentifier(node) && !(ts.getMeaningFromDeclaration(node.parent) & ts.SemanticMeaning.Value) && !(ts.isTypeAliasDeclaration(node.parent) && node.parent.name === node)) {

```

Comment:
Let me see if I can summarise what this means: we now bail for `@callback`, `@typedef` and `@enum` and some (which?) type aliases. Before it looks like there was a check that `@callback` and `@typedef` would usually fail, but sometimes incorrectly pass. But I'm not sure in what circumstances.

Reply:
`getMeaningFromDeclaration` on a `JSDoc` (as is the parent of all of these) isn't `Value` afaik, so those cases are already unreachable.